src/System.IO.FileSystem/tests/FileInfo/Exists.cs

```diff
@@ -130,5 +130,54 @@ public void SymLinksMayExistIndependentlyOfTarget()
             linkPathFI.Refresh();
             Assert.False(linkPathFI.Exists, "linkPath should no longer exist");
         }
+
+        [Fact]
+        public void UnsharedFileExists()
+        {
+            string path = GetTestFilePath();
+            using (FileStream stream = new FileStream(path, FileMode.CreateNew, FileAccess.ReadWrite, FileShare.None))
+            {
+                RemoteInvoke((p) =>
+                {
+                    FileInfo info = new FileInfo(p);
+                    Assert.True(info.Exists);
+                    return SuccessExitCode;
+                }, path).Dispose();
+            }
+        }
+
+        [Fact]
+        public void LockedFileExists()
+        {
+            string path = GetTestFilePath();
+            File.WriteAllBytes(path, new byte[10]);
+
+            using (FileStream stream = new FileStream(path, FileMode.Open, FileAccess.ReadWrite, FileShare.None))
+            {
+                stream.Lock(0, 10);
```

Comment:
This is fine but I notice in the docs for ::LockFile that the length can be longer than the file so I guess Int32.MaxValue would work also..